internal/kube/client.go

```diff
@@ -51,6 +51,35 @@ func NewFromConfig(url, file string) (KymaKube, error) {
 	return NewFromConfigWithTimeout(url, file, defaultHTTPTimeout)
 }
 
+// NewFromRestConfigWithTimeout
+func NewFromRestConfigWithTimeout(config *rest.Config, t time.Duration) (KymaKube, error) {
+
+	config.Timeout = t
+
+	sClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	dClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	istioClient, err := istio.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &client{
+			static:  sClient,
+			dynamic: dClient,
+			istio:   istioClient,
+			restCfg: config,
```

Comment:
For the kubeconfig field try to return not a zero value here (nil), but rather try to construct an empty/dummy kubeconfig and return that.